# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/meson-cc-tests/ipc_rmid_deferred_release.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/meson-cc-tests/ipc_rmid_deferred_release.c
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/cairo/cairo/meson-cc-tests/ipc_rmid_deferred_release.c	Mon May 06 19:25:12 2024 +0000
@@ -0,0 +1,18 @@
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/shm.h>
+int main()
+{
+    char *shmaddr;
+    int id = shmget (IPC_PRIVATE, 4, IPC_CREAT | 0600);
+    if (id == -1) return 2;
+    shmaddr = shmat (id, 0, 0);
+    shmctl (id, IPC_RMID, 0);
+    if ((char*) shmat (id, 0, 0) == (char*) -1) {
+  shmdt (shmaddr);
+  return 1;
+    }
+    shmdt (shmaddr);
+    shmdt (shmaddr);
+    return 0;
+}